Sources/WendyShared/DeviceModels.swift

```diff
@@ -47,36 +191,35 @@ public struct DevicesCollection: Encodable, Sendable {
     }
 
     public func toHumanReadableString() -> String {
-        var results = [String]()
+        let grouped = groupedDevices()
 
-        // Add USB devices section
-        if !usbDevices.isEmpty {
-            var result = "\nUSB Devices:"
-            for device in usbDevices {
-                result += "\n- " + device.toHumanReadableString()
-            }
-            results.append(result)
+        if grouped.isEmpty {
+            return "No devices found."
         }
 
-        // Add Ethernet devices section
-        if !ethernetDevices.isEmpty {
-            var result = "\nEthernet Interfaces:"
-            for device in ethernetDevices {
-                result += "\n- " + device.toHumanReadableString()
+        var result = ""
+
+        for (deviceName, interfaces) in grouped {
+            if !result.isEmpty {
+                result += "\n"
+            }
+
+            // Get the first available agent version (should be the same across interfaces)
+            let agentVersion = interfaces.first(where: { $0.agentVersion != nil })?.agentVersion
+
+            // Add device name with agent version if available
+            result += "\n\(deviceName)"
+            if let version = agentVersion {
+                result += " (Agent: \(version))"
```

Comment:
```suggestion
                result += " (version: \(version))"
```

I think the term "agent" is a bit of an internal detail now